pkg/odo/cli/dev/dev.go

```diff
@@ -127,6 +139,33 @@ func (o *DevOptions) Validate(ctx context.Context) error {
 		}
 		scontext.SetPlatform(ctx, o.clientset.PodmanClient)
 	}
+
+	if o.randomPortsFlag && o.portForwardFlag != nil {
+		return errors.New("--random-ports and --port-forward cannot be used together")
+	}
+
+	if o.portForwardFlag != nil {
```

Comment:
I know that Podman support for this will come after this PR, but until this is done, I think we should either error out if `platform` is `podman` and `--port-forward` is set. Or at least display a warning.
Otherwise, the output is a bit confusing IMO.

```
$ odo dev --debug --port-forward 8089:3000 --port-forward 5005:runtime:5858 --platform podman
  __
 /  \__     Developing using the "my-nodejs" Devfile
 \__/  \    Platform: podman
 /  \__/    odo version: v3.8.0
 \__/

↪ Running on podman in Dev mode
 ✓  Deploying pod [16s]
 ✓  Building your application in container (command: install) [3s]
 •  Executing the application (command: debug)  ...
 -  Forwarding from 127.0.0.1:20001 -> 3000
 -  Forwarding from 127.0.0.1:20002 -> 5858

↪ Dev mode
 Status:
 Watching for changes in the current directory /home/asoro/work/tmp/6479-allow-users-define-custom-port-mappings-for-port-forwarding

 Keyboard Commands:
[Ctrl+c] - Exit and delete resources from podman
     [p] - Manually apply local changes to the application on podman

```

Reply:
I am planning to implement this for podman before our next release, but you're right, it is better to add a warning until them.